### ISSUE TITLE
Incorrect LicenseRef-scancode being used

### ISSUE THREAD

**Issue by ariel11** [clearlydefined/service#1259]
ClearlyDefined picked an incorrect License Ref ID for this component - https://clearlydefined.io/definitions/npm/npmjs/@ag-grid-enterprise/status-bar/30.2.1/30.2.1. 

I don't think another company’s EULA should get “LicenseRef-scancode-warranty-disclaimer,” which is described as a “catch all license” often used by Microsoft in its source code. 

 
![image](https://github.com/user-attachments/assets/703354b4-cd0a-4b94-b138-959594fd8a74)

However, there does not appear to be a License Ref for the AG GRID ENTERPRISE EULA.  How can I nominate that for a LicenseRef-scancode ID?

Also, I hope ClearlyDefined is not defaulting to the closest LicenseRef-scancode ID it thinks it has? If there's not a match, ClearlyDefined should still put NOASSERTION, correct?

**Comment by ariel11** [comment]
@elrayle - FYI. Thank you!

@capfei and @AE49 - FYI

**Comment by elrayle** [comment]
If you have questions or improvements to the licenses coming from ScanCode, you can see your options for support in the [scancode-licensedb support page](https://scancode-licensedb.aboutcode.org/help.html#support).

Screen shot here for reference.  If you go to the support page, there are links.

![scancode support](https://github.com/user-attachments/assets/8acf23ac-b268-492d-92be-a6b67f932121)

**Comment by elrayle** [comment]
@ariel11 I added an issue to scancode-licensedb using primarily your description in this issue.

* https://github.com/aboutcode-org/scancode-licensedb/issues/65

There is another issue in scancode related to whether "unknown" licenses should be NOASSERTION.  The issue has a long discussion, so the link jumps ahead to a comment that is more closely related.

* unexpected LicenseRefs [scancode-toolkit/issue #3974](https://github.com/aboutcode-org/scancode-toolkit/issues/3974#issuecomment-2491785295)
----

NOTE: There still may be work on the CD side to address:

> I hope ClearlyDefined is not defaulting to the closest LicenseRef-scancode ID it thinks it has? If there's not a match, ClearlyDefined should still put NOASSERTION, correct?

**Comment by bduranc** [comment]
Sorry to jump late into this.

But isn't _[LicenseRef-scancode-unknown](https://scancode-licensedb.aboutcode.org/unknown.html)_ basically the same as what we would traditionally call a NOASSERTION?

`This applies to the case where there is text referring to a license, but it is not possible to determine exactly which license.`

**Comment by bduranc** [comment]
> Also, I hope ClearlyDefined is not defaulting to the closest LicenseRef-scancode ID it thinks it has? If there's not a match, ClearlyDefined should still put NOASSERTION, correct?

I agree with @ariel11's initial concern around the use of _LicenseRef-scancode-warranty-disclaimer_ in this particular example.
I'm still familiarizing myself with the LicenseRefs available, but it would seem appropriate that cases where a LicenseRef cannot be confidently assigned, default to NOASSERTION (or [LicenseRef-scancode-unknown](https://scancode-licensedb.aboutcode.org/unknown.html) assuming it's a valid equivalent).

**Comment by ariel11** [comment]
I agree @bduranc. Also, there seems to be _two_ LicenseRef's that both mean "there appears to be a LICENSE or license info, but can't tell what it is" - _LicenseRef-scancode-unknown-license-reference_ **and** _LicenseRef-scancode-unknown_. Since users of ClearlyDefined will have established processes around NOASSERTION, it seems prudent to have ClearlyDefined continue using NOASSERTION in these cases. Otherwise, folks will have to update their processes (if they want to) to investigate _three_ "can't tell what the license is" ID's. That is doable, but noisier.

**Comment by bduranc** [comment]
> Since users of ClearlyDefined will have established processes around NOASSERTION, it seems prudent to have ClearlyDefined continue using NOASSERTION in these cases. Otherwise, folks will have to update their processes (if they want to) to investigate three "can't tell what the license is" ID's. That is doable, but noisier.

100% agreed. I think this would have the least impact on our downstream users and still allow them to benefit from LicenseRef for everything else.

So basically, to codify the logic:
```
if is_known_SPDX_license then:
    assign SPDX_ID
elif is_known_license_ref:
    assign license_ref
else:
    assign NOASSERTION
```
